### PR TITLE
Add example visual regression test using jest

### PR DESCRIPTION
**Pull request by swissspidy**
Took #1170 as an inspiration to test visual regression testing using [`jest-image-snapshot`](https://github.com/americanexpress/jest-image-snapshot).

Used #1299 as an example of something that can benefit from this.

The result is a snapshot of the truncated text that is now compared against with every test run.

layer-js-layer-text-should-show-ellipsis-for-overflowing-text-1-snap

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -132,7 +132,7 @@ jobs:
     - stage: test
       name: JavaScript unit tests
       php: 7.4
-      env: JS=1
+      env: JS=1 PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=
```
Comment (swissspidy) [inline .travis.yml]: Maybe create a separate job just for visual regression tests?

**Comment by swissspidy** [comment]
> Expected image to match or be a close match to snapshot but was 0.12958333333333333% different from snapshot (622 differing pixels).
    See diff for details: `/home/travis/build/google/web-stories-wp/assets/src/edit-story/components/panels/layer/test/__image_snapshots__/__diff_output__/layer-js-layer-text-should-show-ellipsis-for-overflowing-text-1-diff.png`

strange 🤷

**Comment by dvoytenko** [comment]
The main issue as we touched on online is that we have to do a screenshot-per-platform since folk are on different hardware. Even device pixel ratio, presence of a mouse, and system settings could affect this.

**Comment by swissspidy** [comment]
In some cases per-platform screenshots make sense, which is where a tool like Percy might be useful.

I don't think we need that for simple tests like the one in this PR though. For consistent results we could run these image snapshot tests in a Docker container. I'd want to run visual regression tests in a separate job anyway, which would make that easier. It's also what the `jest-image-snapshot` contributors recommend.

Thoughts?

**Comment by dvoytenko** [comment]
> In some cases per-platform screenshots make sense, which is where a tool like Percy might be useful.
> 
> I don't think we need that for simple tests like the one in this PR though. For consistent results we could run these image snapshot tests in a Docker container. I'd want to run visual regression tests in a separate job anyway, which would make that easier. It's also what the `jest-image-snapshot` contributors recommend.
> 
> Thoughts?

I think the issue will be that when a correct change is required, anyone should be able to regenerate the "golden" screenshot. As long as it's possible, easy, kind-of-fast, and is stable - then we can do it. Otherwise, I'd just wait until we can do something like Percy.

**Comment by swissspidy** [comment]
Closing in favor of #1596.

**Pull request closed without merging**